python-lib/ontology_tagging/formatter_by_document.py

```diff
@@ -0,0 +1,442 @@
+
+        Returns:
+            - Tuple(dict,dict) : The enriched dictionaries 'line' and 'line_full'
+
+        """
+        keyword = match.text
+        tag = match.ent_id_
+        category = match.label_
+        sentence = sentence
+        if tag not in line_full[category]:
+            line_full[category][tag] = {
+                "count": 1,
+                "sentences": [sentence],
+                "keywords": [keyword],
+            }
+
+            line[category].append(tag)
+        else:
+            line_full[category][tag]["count"] += 1
+            if sentence not in line_full[category][tag]["sentences"]:
+                line_full[category][tag]["sentences"].append(sentence)
+            if keyword not in line_full[category][tag]["keywords"]:
+                line_full[category][tag]["keywords"].append(keyword)
+        return line, line_full
+
+    def _merge_df_columns_category(
+        self, input_df: pd.DataFrame, text_column: AnyStr
+    ) -> pd.DataFrame:
+        """
+        Called by _write_row_category, when the format is 'one row per document (array columns)' and there are categories.
+        Insert columns 'tag_keywords' and 'tag_sentences' into self.output_df
+
+        Returns:
+            pd.DataFrame: the complete output dataframe after setting the columns in the right order
+
+        """
+        tag_list_columns = self.output_df.columns.tolist()
+        tag_list_columns_unique = generate_unique_columns(
+            df=self.output_df,
+            columns=[
+                unicode_normalize_text(text=column) for column in tag_list_columns
+            ],
+            prefix="tag_list",
+        )
+        self.output_df.columns = tag_list_columns_unique
+        for column, column_unique in zip(tag_list_columns, tag_list_columns_unique):
+            category = column.split("_")[-1]
+            self.column_descriptions[column_unique] = f"List of '{category}' tags"
+        self.output_df.insert(
+            len(self.output_df.columns),
+            self.tag_columns[0],
+            self.tag_keywords,
+            True,
+        )
+        self.output_df.insert(
+            len(self.output_df.columns),
+            self.tag_columns[1],
+            self.tag_sentences,
+            True,
+        )
+        self.tag_columns = tag_list_columns_unique + self.tag_columns
+        return self._set_columns_order(input_df, self.output_df, text_column)
+
+    def _merge_df_columns(
+        self, input_df: pd.DataFrame, text_column: AnyStr
+    ) -> pd.DataFrame:
+        """
+        Called by write_df
+        Insert columns 'tag_sentences' and 'tag_keywords' into self.output_df and there is no category
+
+        Returns:
+            pd.DataFrame: the complete output dataframe after setting the columns in the right order
+
+        """
+        for column in self.tag_columns[::-1]:
+            self.output_df.set_index(column, inplace=True)
+            self.output_df.reset_index(inplace=True)
```

Comment:
I don't understand what this is doing

Reply:
Actually it is not usefull anymore, I got rid of it in PR #21